commons-core/src/test/scala/com/avsystem/commons/serialization/json/JsonStringInputOutputTest.scala

```diff
@@ -0,0 +1,206 @@
+package com.avsystem.commons
+package serialization.json
+
+import com.avsystem.commons.serialization.GenCodec.ReadFailure
+import com.avsystem.commons.serialization.{GenCodec, Input, Output}
+import org.scalatest.{FunSuite, Matchers}
+
+import scala.collection.mutable.ListBuffer
+import scala.util.Random
+
+class JsonStringInputOutputTest extends FunSuite with SerializationTestUtils with Matchers {
+
+  import JsonStringInput.read
+  import JsonStringOutput.write
+
+  test("raw json list test") {
+    val jsons = List("123", "null", "\"str\"", "4.5", "[1,2,3]", "{\"a\": 123, \"b\": 3.14}")
+    val sb = new JStringBuilder
+    val output = new JsonStringOutput(sb)
+    val lo = output.writeList()
+    jsons.foreach(json => lo.writeElement().writeRawJson(json))
+    lo.finish()
+    val jsonList = sb.toString
+
+    assert(jsonList == jsons.mkString("[", ",", "]"))
+
+    val input = new JsonStringInput(new JsonReader(jsonList))
+    val li = input.readList()
+    val resBuilder = new ListBuffer[String]
+    while (li.hasNext) {
+      resBuilder += li.nextElement().readRawJson()
+    }
+    assert(resBuilder.result() == jsons)
+  }
+
+  test("raw json object test") {
+    val jsons = IListMap("a" -> "123", "b" -> "null", "c" -> "\"str\"", "d" -> "4.5", "e" -> "[1,2,3]", "f" -> "{\"a\": 123, \"b\": 3.14}")
+    val sb = new JStringBuilder
+    val output = new JsonStringOutput(sb)
+    val oo = output.writeObject()
+    jsons.foreach { case (fn, fv) => oo.writeField(fn).writeRawJson(fv) }
+    oo.finish()
+    val jsonList = sb.toString
+
+    assert(jsonList == jsons.map({ case (k, v) => s""""$k":$v""" }).mkString("{", ",", "}"))
+
+    val input = new JsonStringInput(new JsonReader(jsonList))
+    val oi = input.readObject()
+    val resBuilder = IListMap.newBuilder[String, String]
+    while (oi.hasNext) {
+      val fi = oi.nextField()
+      resBuilder += ((fi.fieldName, fi.readRawJson()))
+    }
+    assert(resBuilder.result() == jsons)
+  }
+
+  def roundtrip[T: GenCodec](name: String)(values: Seq[T]): Unit = {
```

Comment:
Take an implicit `org.scalactic.source.Position` to this function

Reply:
Agreed to the first, but I don't want to just pass a single value, because I don't care for sequence ser/de here.

Reply:
If I understand your intention well then I think varargs would be more appropriate than `Seq`.